### PR TITLE
[ISSUE-363] CSI Operator fixes (found with e2e)

### PR DESCRIPTION
**Pull request by safronovD**
### Issue dell/csi-baremetal#363

- CSI CR scope -> Namespaced
- Update scheduler rbac
- Fix use-node-annotaion feature
- Changes from  https://github.com/dell/csi-baremetal/pull/369

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Testing details_

### CODE REVIEW

File: charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml [hunk 1]
```diff
@@ -2,13 +2,13 @@ apiVersion: csi-baremetal.dell.com/v1
...
 metadata:
   name: {{ .Release.Name }}
-  namespace: {{ .Release.Namespace }}
   labels:
     app.kubernetes.io/name: csi-baremetal
     app.kubernetes.io/managed-by: csi-baremetal-operator
     app.kubernetes.io/version: 1.0.0
 spec:
   platform: {{ .Values.platform }}
+  namespace: {{ .Release.Namespace }}
```
Comment (AnastasiaShemyakinskaya) [inline charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml:11]: Why do we need namespace in spec?

File: test/e2e/common/deploy.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		if err := c.CoreV1().Namespaces().Delete(chart.namespace, nil); err != nil {
+			e2elog.Logf("Namespace deletion failed. Namespace: %s", chart.namespace)
+		}
+	}
+
+	if err := executor.InstallRelease(&chart, installArgs); err != nil {
+		return nil, err
+	}
+
+	time.Sleep(sleepBeforeWait)
```
Comment (AnastasiaShemyakinskaya) [inline test/e2e/common/deploy.go:57]: Why do we need to sleep before polling pods?

File: charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml (hunk 1 above)
Comment (safronovD) [reply charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml:11]: We created  csi-deployment CRD as cluster-scoped, so namespace in meta section was empty ever (even with set value). And if we want to create controlled resources in a custom namespace, we should pass this parameter another way

File: test/e2e/common/deploy.go (hunk 2 above)
Comment (safronovD) [reply test/e2e/common/deploy.go:57]: I had lags using `go test` in cli sometimes, when helm has already returned response, but `WaitForPodsRunningReady` has shown 0/0 pods and 0 replicasets in namespace. I have no ideas how to deal with it without Sleep

File: charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml:11]: Shouldn't csi-deployment be namespaced? Correct me if I'm wrong

Comment (safronovD) [reply charts/csi-baremetal-deployment/templates/csi-baremetal_v1_deployment.yaml]: Csi-deployment scope was changed to namespaced

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+CSI Baremetal operator e2e testing
+---------------------
+
+Last update: 19.04.2021
+
+! Kind-cluster has pre-loaded csi images
```
Comment (mishoyama) [inline test/README.md:6]: Please remove `!` symbol

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+1. Set CSI version
+
+```
+export csiVersion=...
```
Comment (mishoyama) [inline test/README.md:11]: Environment variables are defined with capital letters usually. Please use the following format for all env vars:
`export CSI_VERSION=<csi version>`

File: test/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    
+```
+make docker-build
+make kind-load
+```
+
+3. Set Operator version
+
+```
+export operatorVersion=...
```
Comment (mishoyama) [inline test/README.md:24]: same here

File: test/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+4. Copy actual charts from csi-baremetal-operator repo
+
+```
+cp -r ./charts /tmp/charts
```
Comment (mishoyama) [inline test/README.md:30]: Does it make sense to add a target to Makefile for CI tests?

File: test/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+CSI Baremetal operator e2e testing
+---------------------
```
Comment (mishoyama) [inline test/README.md:2]: Please move file to `docs` folder

File: test/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+4. Copy actual charts from csi-baremetal-operator repo
+
+```
+cp -r ./charts /tmp/charts
+```
+
+5. Run e2e testing
+
+```
+CI=true go test -v test/e2e/csi_operator_e2e_test.go -ginkgo.v -ginkgo.progress -kubeconfig={$HOME}/.kube/config -operatorVersion={$operatorVersion} -csiVersion={$csiVersion}
```
Comment (mishoyama) [inline test/README.md:36]: Is passing Operator version is really necessary? Can't it be set when preparing charts?

File: test/e2e/common/deploy.go [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+func DeployOperator(c clientset.Interface) (func(), error) {
```
Comment (mishoyama) [inline test/e2e/common/deploy.go:33]: Please add comments to all exported methods. Doesn't linter complain about this?

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: test/README.md (hunk 3 above)
Comment (safronovD) [reply test/README.md]: done

File: test/README.md (hunk 4 above)
Comment (safronovD) [reply test/README.md]: done

File: test/README.md (hunk 5 above)
Comment (safronovD) [reply test/README.md]: done

File: test/README.md (hunk 7 above)
Comment (safronovD) [reply test/README.md]: moved and renamed with `e2e_testing_on_kind`

File: test/README.md (hunk 8 above)
Comment (safronovD) [reply test/README.md]: Is reading version from env a suitable way?
```
operatorVersion = os.Getenv(operatorVersionEnv)
```

File: test/e2e/common/deploy.go (hunk 9 above)
Comment (safronovD) [reply test/e2e/common/deploy.go]: added to all public methods

Comment (safronovD) [reply test/e2e/common/deploy.go]: But yes, linter had no issues

**Review by mishoyama — APPROVED** [review]

**Review by yurkov-anton — APPROVED** [review]

**Review by painhardcore — APPROVED** [review]

**Pull request merged**